activerecord/lib/active_record/associations/preloader/association.rb

```diff
@@ -44,7 +44,7 @@ def associate_records_to_owner(owner, records)
             association = owner.association(reflection.name)
             association.loaded!
             if reflection.collection?
-              association.target.concat(records)
+              association.target = records
```

Comment:
Isn't this line changing code behaviour? Whereas before the code was adding records to `target`, we are now _overwriting_ the target with records. Is that intentional?

Reply:
This is intentional because we don't want the inverse in a preloaded `has_many` otherwise we introduce duplicates and the test suite turns red. I thought if this needed to concat, there would be a test for it, but there doesn't seem to be.

Reply:
Ah, I see. Thanks for the explanation!